app/assets/javascripts/discourse/app/initializers/narrow-desktop.js

```diff
@@ -0,0 +1,45 @@
+import NarrowDesktop from "discourse/lib/narrow-desktop";
+
+export default {
+  name: "narrow-desktop",
+
+  initialize(container) {
+    NarrowDesktop.init();
+    let site;
+    if (!container.isDestroyed) {
```

Comment:
I wonder if we should just skip this in test entirely and just rely on setting `site.narrowDesktopView` manually. Might be why your tests are flaky.